cmd/main.go

```diff
@@ -335,12 +343,24 @@ func startDash0Controller(
 		setupLog.Info("Dash0 webhooks have been disabled via configuration.")
 	}
 
+	cfg := ctrl.GetConfigOrDie()
+	startupInstrumenterK8sClient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return err
+	}
+	startupInstrumenter := &instrumentation.Instrumenter{
+		Client:               startupInstrumenterK8sClient,
+		Clientset:            clientset,
+		Recorder:             mgr.GetEventRecorderFor("dash0-controller"),
+		Images:               images,
+		OTelCollectorBaseUrl: oTelCollectorBaseUrl,
+	}
 	go func() {
 		time.Sleep(10 * time.Second)
```

Comment:
The sleep is a workaround for the following issue: 
* We need to kick this (`InstrumentAtStartup`) off before we call [mgr.Start()](https://github.com/dash0hq/dash0-operator/blob/main/cmd/main.go#L273), because `mgr.Start()` blocks.
* If `InstrumentAtStartup` runs right aways, before `mgr.Start()` has actually run, it will throw an error, since the Kubernetes `client` that we get from the `mgr` is not started yet.

As the PR description explains, the aim of this PR is get rid of this arbitrary 10 second sleep -- by using a dedicated Kubernetes client that is independent from `mgr`. I was working on that when you threw your latest PR over the wall :-)

Reply:
The sleep is gone now, as well as the goroutine -- which was the ultimate goal of the PR.